pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StEqualsFunction.java

```diff
@@ -53,14 +54,22 @@ public void init(List<TransformFunction> arguments, Map<String, DataSource> data
     TransformFunction transformFunction = arguments.get(0);
     Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
         "First argument must be single-valued for transform function: %s", getName());
-    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES,
-        "The first argument must be of bytes type");
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        String.format("The %s argument must be of type %s , but was %s",
+            "first",
```

Comment:
nit, you can inline `first` in the msg

Reply:
This is worthwhile because it avoids the varargs overload too